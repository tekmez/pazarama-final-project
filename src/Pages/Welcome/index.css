.welcome-top-container{
    display: flex;
}
.welcome-text-container{
    width: 650px;
    height: 440px;
    margin-right: 10em;
    border-radius: 25px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-evenly;
    background: rgba( 255, 255, 255, 0.25 );
    box-shadow: 0 8px 32px 0 rgba( 31, 38, 135, 0.37 );
    backdrop-filter: blur( 4.5px );
    -webkit-backdrop-filter: blur( 4.5px );
}
.welcome-btn-container{
    width: 250px;
    height: 440px;
    border-radius: 25px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-evenly;
    background: rgba( 255, 255, 255, 0.25 );
    box-shadow: 0 8px 32px 0 rgba( 31, 38, 135, 0.37 );
    backdrop-filter: blur( 4.5px );
    -webkit-backdrop-filter: blur( 4.5px );
}
.welcome-title, .welcome-text{
    width: 450px;
    height: 50px;
    /* background-color: #fff; */
    border-radius: 15px;
    text-align: center;
}
.welcome-text{
    font-size: 20px;
    height: 150px;
    display: flex;
    align-items: center;
    line-height: 30px;
}
.medical{
    position: absolute;
    margin-left: 15px;
}
.search{
    position: absolute;
    margin-left: 25px;
}
.admin{
    position: absolute;
    margin-left: 55px;
}