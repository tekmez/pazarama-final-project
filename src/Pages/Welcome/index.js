import React, { useEffect } from "react";
import { useNavigate } from "react-router-dom";
import {
  faFileMedical,
  faSearch,
  faUserShield,
} from "@fortawesome/free-solid-svg-icons";
import { useDispatch } from "react-redux";
import { getUser } from "../../Redux/Services";
import Button from "../../components/Button";
import "./index.css";

function Welcome() {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  useEffect(() => {
    dispatch(getUser());
  }, []);
  return (
    <div className="welcome-top-container">
      <div className="welcome-text-container">
        <h1 className="welcome-title">Welcome Application Form</h1>
        <p className="welcome-text">
          You can easily create your application form by using the application
          form I have prepared to use in your job applications.
        </p>
      </div>
      <div className="welcome-btn-container">
        <Button
          text="Create Form"
          icon={faFileMedical}
          classNameBtn="btn"
          className="medical"
          onClick={() => navigate("/CreateForm")}
        />
        <Button
          text="Find Form"
          icon={faSearch}
          classNameBtn="btn"
          className="search"
          onClick={() => navigate("/SearchFormPage")}
        />
        <Button
          text="Admin"
          icon={faUserShield}
          classNameBtn="btn"
          className="admin"
          onClick={() => navigate("/LoginPage")}
        />
      </div>
    </div>
  );
}

export default Welcome;
