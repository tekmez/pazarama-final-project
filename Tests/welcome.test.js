import React from "react";
import { shallow } from "enzyme";
import Welcome from "../src/Pages/Welcome/index";
import renderer from "react-test-renderer";

const wrapper = shallow(<Welcome />)

describe("p", () => {
    it("p", () => {
        expect(wrapper.find("p").text()).toEqual("You can easily create your application form by using the application form I have prepared to use in your job applications.")
    })
    it("h", () => {
        expect(wrapper.find("h1").text()).toEqual("Welcome Application Form")
    })
})

test(" Welcome snap", () => {
  const tree = renderer
    .create(
      <Welcome />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});