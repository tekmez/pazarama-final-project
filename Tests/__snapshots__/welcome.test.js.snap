// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Welcome snap 1`] = `
<div
  className="welcome-top-container"
>
  <div
    className="welcome-text-container"
  >
    <h1
      className="welcome-title"
    >
      Welcome Application Form
    </h1>
    <p
      className="welcome-text"
    >
      You can easily create your application form by using the application form I have prepared to use in your job applications.
    </p>
  </div>
  <div
    className="welcome-btn-container"
  >
    <div
      className="btn-container"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        Create Form
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-file-medical fa-w-12 medical"
          data-icon="file-medical"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M377 105L279.1 7c-4.5-4.5-10.6-7-17-7H256v128h128v-6.1c0-6.3-2.5-12.4-7-16.9zm-153 31V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm64 160v48c0 4.4-3.6 8-8 8h-56v56c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8v-56h-56c-4.4 0-8-3.6-8-8v-48c0-4.4 3.6-8 8-8h56v-56c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v56h56c4.4 0 8 3.6 8 8z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </div>
    <div
      className="btn-container"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        Find Form
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-search fa-w-16 search"
          data-icon="search"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </div>
    <div
      className="btn-container"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        Admin
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-user-shield fa-w-20 admin"
          data-icon="user-shield"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M622.3 271.1l-115.2-45c-4.1-1.6-12.6-3.7-22.2 0l-115.2 45c-10.7 4.2-17.7 14-17.7 24.9 0 111.6 68.7 188.8 132.9 213.9 9.6 3.7 18 1.6 22.2 0C558.4 489.9 640 420.5 640 296c0-10.9-7-20.7-17.7-24.9zM496 462.4V273.3l95.5 37.3c-5.6 87.1-60.9 135.4-95.5 151.8zM224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm96 40c0-2.5.8-4.8 1.1-7.2-2.5-.1-4.9-.8-7.5-.8h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c6.8 0 13.3-1.5 19.2-4-54-42.9-99.2-116.7-99.2-212z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
